File: codes/from_python_to_go/generators.py
#!/usr/bin/python3


def fib(n):
    a, b = 0, 1
    for i in range(n):
        a, b = b, a + b
        yield a

sequence = fib(10)
print(next(sequence))
print(next(sequence))
print(next(sequence))
print(80 * '-')
for number in fib(10):
    print(number)

File: codes/from_python_to_go/generators.go
package main

import "fmt"

func fib(n int) chan int {
	c := make(chan int)
	go func() {
		a, b := 0, 1
		for i := 0; i < n; i++ {
			a, b = b, a+b
			c <- a
		}
		close(c)
	}()
	return c
}

func main() {
	for x := range fib(10) {
		fmt.Println(x)
	}
}
